### PR TITLE
Add identity client methods for hosted broker login flow.

### PR DESCRIPTION
**Pull request by galxy25**

### CODE REVIEW

File: identityClient/api.go [hunk 1]
```diff
@@ -98,7 +98,35 @@ type RegisterRealmBrokerIdentityRequest struct {
...
+// BrokerChallengeRequest wraps Identifying information for the Identity initiating the challenge.
+type BrokerChallengeRequest struct {
+	RealmName string // The name of the realm the Identity is a member of.
+	Action    string `json:"action"`   // The requested broker flow action to perform. Currently only challenge is supported.
+	Username  string `json:"username"` // The name for the Identity initiating the challenge.
+}
+
+// EmailOTP wraps a one time password provided via an email challenge.
+type EmailOTP struct {
+	EmailOTP string // The one-time password from the email challenge issued.
```
Comment (lkwdwrd) [inline identityClient/api.go]: does this need a `json:` tag?

File: identityClient/api.go [hunk 2]
```diff
@@ -98,7 +98,35 @@ type RegisterRealmBrokerIdentityRequest struct {
...
+}
+
+// BrokerLoginRequest proof that the Identity has completed the broker challenge
+// along with key material to encrypt the login response.
+type BrokerLoginRequest struct {
+	RealmName    string            // The name of the realm the Identity is a member of.
+	Action       string            `json:"action"`                // The requested broker flow action to perform. Currently only login is supported
+	NoteID       uuid.UUID         `json:"note_id"`               // The ID of the recovery Note the email challenge was for.
+	PublicKeys   map[string]string `json:"public_key"`            // The public key to use to encrypt the recovery note.
+	SigningKeys  map[string]string `json:"signing_key,omitempty"` // The signing key to use to verify the integrity of the recovery note.
```
Comment (lkwdwrd) [inline identityClient/api.go]: Because this is notes, this should never be empty -- any reason to `omitempty`?

Comment (galxy25) [reply identityClient/api.go]: good catch, copy pasta

File: identityClient/api.go (hunk 1 above)
Comment (galxy25) [reply identityClient/api.go]: good catch

**Pull request merged**